sam.c

```diff
@@ -522,6 +522,7 @@ int sam_index_build2(const char *fn, const char *fnidx, int min_shift)
         idx = bam_index(fp->fp.bgzf, min_shift);
         if (idx) {
             ret = hts_idx_save_as(idx, fn, fnidx, (min_shift > 0)? HTS_FMT_CSI : HTS_FMT_BAI);
+            ret = ret ? -4 : 0;
```

Comment:
Essentially yes.  It's the analogue of if (ret != 0) ret = -4, but basically it's the same thing as the only working return value is zero.

I don't particularly find ?: incomprehensible, but if you feel strongly about it I can resubmit the patch with the change you suggest.